src/iso.c

```diff
@@ -274,9 +275,12 @@ static BOOL check_iso_props(const char* psz_dirname, int64_t file_length, const
 			}
 		}
 
-		// Check for ReactOS' setupldr.sys anywhere
-		if ((img_report.reactos_path[0] == 0) && (safe_stricmp(psz_basename, reactos_name) == 0))
-			static_strcpy(img_report.reactos_path, psz_fullpath);
+		// Check for ReactOS presence
+		if (img_report.reactos_path[0] == 0) {
+			if ((safe_stricmp(psz_basename, reactos_name) == 0) ||
+				(safe_stricmp(psz_basename, reactos_old_name) == 0))
+					static_strcpy(img_report.reactos_path, psz_fullpath);
```

Comment:
Since there was a comment suggesting a problem with using freeldr.sys in (probably older?) ReactOS ISOs. Maybe it could make sense to ensure that setupldr.sys is always used whenever it exists, and check for it until it is found or we went through all files on ISO.

```suggestion
		if ((safe_strstr(img_report.reactos_path, reactos_old_name) == NULL) &&
			((safe_stricmp(psz_basename, reactos_name) == 0) ||
			(safe_stricmp(psz_basename, reactos_old_name) == 0)))
			static_strcpy(img_report.reactos_path, psz_fullpath);
```

Reply:
The problem was that freeldr.sys file was absent from livecd, only setupldr was there, but both files have the same content, so it should be fine ...

Now, I should add, that in much older reactors versions (like the 0.3.x ones), setupldr contained only the startup code to start the bootcd usetup, and it was different from the regular freeldr used in the livecd or in a regular ReactOS installation on a HDD.